src/ipc/ipc4/dai.c

```diff
@@ -240,6 +240,59 @@ void dai_dma_release(struct comp_dev *dev)
 	}
 }
 
+static int dai_init_llp_info(struct comp_dev *dev)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct ipc4_copier_module_cfg *copier_cfg;
+	union ipc4_connector_node_id node;
+	uint32_t llp_slot_size;
+	uint32_t max_slot;
+	uint32_t node_id;
+	uint32_t offset;
+	int i;
+
+	copier_cfg = dd->dai_spec_config;
+	node = copier_cfg->gtw_cfg.node_id;
+
+	/* HDA doesn't use llp slot */
+	if (dd->ipc_config.type == SOF_DAI_INTEL_HDA)
+		return 0;
+
+	/* don't support more gateway like EVAD */
+	if (node.f.dma_type >= ipc4_max_connector_node_id_type) {
+		comp_err(dev, "unsupported gateway %d", (int)node.f.dma_type);
+		return -EINVAL;
+	}
+
+	dd->slot_info.node_id = node.dw & IPC4_NODE_ID_MASK;
+	llp_slot_size = sizeof(struct ipc4_llp_reading_slot);
+
+	/* sdw with multiple gateways uses sndw_reading_slots */
+	if (node.f.dma_type == ipc4_alh_link_output_class && node.f.v_index >= ALH_MULTI_GTW_BASE) {
+		offset = SRAM_REG_LLP_SNDW_READING_SLOTS;
+		max_slot = IPC4_MAX_LLP_SNDW_READING_SLOTS - 1;
+	} else {
+		offset = SRAM_REG_LLP_GPDMA_READING_SLOTS;
+		max_slot = IPC4_MAX_LLP_GPDMA_READING_SLOTS;
+	}
+
+	/* find unused llp slot with node_id of zero */
+	for (i = 0; i < max_slot; i++, offset += llp_slot_size) {
+		node_id = mailbox_sw_reg_read(offset + i * llp_slot_size);
```

Comment:
this should be just `offset` now of course

Reply:
sure, thanks.updated

Reply:
@RanderWang sorry, I still see the same

Reply:
@lyakh  "find unused llp slot offset" is not what your want ? And what do you suggest ? Thanks!

Reply:
> @lyakh "find unused llp slot offset" is not you want ? And what do you suggest ? Thanks!

@RanderWang line 281 should now be
```
		node_id = mailbox_sw_reg_read(offset);
```

Reply:
sure, I lost myself. thanks!

Reply:
last time I changed the offset according to your comments but didn't realized that I should also change here.